lib/safer_rails_console/patches/sandbox/auto_rollback.rb

```diff
@@ -4,21 +4,16 @@ module SaferRailsConsole
   module Patches
     module Sandbox
       module AutoRollback
-
-        def self.rollback_and_begin_new_transaction
-          connection = ::ActiveRecord::Base.connection
-          connection.rollback_db_transaction
-          connection.begin_db_transaction
-        end
-
-        def self.handle_and_reraise_exception(error)
-          if error.message.include?('PG::ReadOnlySqlTransaction')
+        def self.on(error, message:)
```

Comment:
What's the reason for naming this `on`? `handle_and_reraise_exception` feels more explicit.

Reply:
It felt like an easier to read API:
`SaferRailsConsole::Patches::Sandbox::AutoRollback.on(e, message: 'PG::ReadOnlySqlTransaction')`
but I can revert back

Reply:
It's just not obvious to me that `#on` would have anything to do with error handling, so more verbose feels better than less. But up to you.

Reply:
Changed back since it wouldn't be backward compatible.
Also, if the change added confusion instead of readability then its purpose was defeated 😅